frontend/src/api/index.ts

```diff
@@ -130,7 +130,7 @@ class RequestHttp {
                                 response.status,
                                 response.data && response.data['message'] ? response.data['message'] : '',
                             );
-                            return;
+                            return Promise.reject(error);
                         default:
                             globalStore.isLogin = false;
                             globalStore.errStatus = 'code-' + response.status;
```

Comment:
There is one potential issue with handling the `default` case. The current implementation will not reject the promise if the status code does not match any of the specified cases (e.g., 401, 403) but still returns the message from the response data. To make this consistent, you should handle all error codes similarly and then resolve or reject the promise appropriately.

Additionally, adding a more descriptive description for the error would be helpful. Here's an updated version:

```javascript
@@ -130,7 +130,7 @@ class RequestHttp {
                                 response.status,
                                 response.data && response.data['message'] ? response.data['message'] : 'API request failed',
                             );
-                            return;
+                            return Promise.reject(new Error(`Request to ${url} failed with status ${response.status}: ${error.message}`));
                         default:
                             globalStore.isLogin = false;
                             globalStore.errStatus = 'code-' + response.status;
```

This ensures that the promise is handled consistently, providing a clear failure message when there's an unexpected status code.